apps/api/src/controllers/applicationController.ts

```diff
@@ -210,3 +210,47 @@ export const rejectApplication = async ({ applicationId }: { applicationId: numb
 		return failure(message, error);
 	}
 };
+
+export const submitRevision = async ({ applicationId }: { applicationId: number }) => {
+	try {
+		// Fetch application
+		const database = getDbInstance();
+		const service: ApplicationService = applicationSvc(database);
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) {
+			return result;
+		}
+
+		const application = result.data;
+
+		const appStateManager = new ApplicationStateManager(application);
+
+		if (
+			appStateManager.state === ApplicationStates.DAC_REVISIONS_REQUESTED ||
+			appStateManager.state === ApplicationStates.INSTITUTIONAL_REP_REVISION_REQUESTED
+		) {
+			return failure('Application revision is already submitted.', 'RejectionConflict');
+		}
+
+		let submittedRevision;
+		if (appStateManager.state === ApplicationStates.DAC_REVIEW) {
+			submittedRevision = await appStateManager.submitDacRevision();
+		} else {
+			submittedRevision = await appStateManager.submitRepRevision();
+		}
+
+		if (!submittedRevision.success) {
+			return failure(submittedRevision.message || 'Failed to submit application revision.', 'StateTransitionError');
+		}
+
+		const updatedResult = await service.getApplicationById({ id: applicationId });
+
+		return updatedResult;
```

Comment:
`submitDacRevision` and `submitRepRevision` both perform an application db update so `updatedResult` can be removed
You can change `submittedRevision` to `revisionResult` and return that instead